server.js

```diff
@@ -17,115 +26,119 @@ const options = {
+    // .then(OneMoreAsyncFunc) (something) => {return somethingElse}
+    // .catch(errorHandlingFunction)
+
+  // callback hell is a real thing.  Promises solves this.
+//===========================
+
+//ORIGINAL SERVER REQUESTS before 3-21-19
 server.use(express.json()) // bodyParser function for json payloads
 
 server.use(helmet())
-server.use(cors()); // ie between netlify, heroku and mlab
 
-// const memCache ={}; 
+// Allow Cross-origin Resource Sharing i.e. between netlify, heroku and mlab
+// server.use(cors());
+server.use((req, res, next) =>{
+  res.header("Access-Control-Allow-Origin", "*");  //`*` allows all sites to make requests.  change to specific domains to restrict access.
+  res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept, Authorization");
+  if (req.method === 'OPTIONS') {
+    res.header('Access-Control-Allow-Methods', 'PUT, POST, PATCH,DELETE, GET')
+    return res.status(200).json({})
+  }
+  next();  // allows other routes to take over.
+})
 
 //add CRUD routes
 server.get('/', (req, res) => {
-  res.send('Hello from the express server'); // sanity check
+  res.send('Hello from the d2rd Notes back-end express server using MongoDB on mLab'); // sanity check
 });
 
-server.get('/ElectricUprights', (req, res) => {
-  ElectricUpright.find()
+server.get('/Notes', (req, res) => {
+  Note.find()
     .then((data) => {
       res.json(data)
     })
     .catch(err => console.log(err.message))
 })
 
-server.post('/ElectricUprights/create', (req, res) => {
-  const { title, priority, body, price, itemURL, reviewURL } = req.body;
-  const myNote = { title, priority, body, price, itemURL, reviewURL };
-  const newNote = new ElectricUpright(myNote)
+server.post('/Notes/create', (req, res) => {
+  const { title, body } = req.body;
+  const myNote = { title, body };
+  const newNote = new Note(myNote)
   newNote.save()
     .then(note => {
       res.status(201).json(note)
     })
     .catch(err => console.log(err))
 });
-// METHOD 1 req.body
-// server.put('/ElectricUprights/update/', (req, res) => {
-//   console.log(req.body)
-//   ElectricUpright.findByIdAndUpdate(req.body._id, {price: req.body.price, itemURL: req.body.itemURL})
-//     .then(note => {
-//       res.status(201).json(note)
-//     })
-//     .catch(err => console.log(err))
-// })
-// METHOD 2 req.params
-server.put('/ElectricUprights/update/:id', (req, res) => {
+
+server.put('/Notes/update/:id', (req, res) => {
   console.log(req.params.id)
-  ElectricUpright
-    .findByIdAndUpdate(req.params.id, {price: req.body.price, itemURL: req.body.itemURL})
+  Note
+    .findByIdAndUpdate(req.params.id, {title: req.body.title, body: req.body.body})
     .then(note => {
       res.status(201).json(note)
     })
     .catch(err => console.log(err))
 })
 
-server.delete('/ElectricUprights/delete/:id', deleteFunc)
+server.delete('/Notes/delete/:id', deleteFunc)
 
 function deleteFunc (req, res) {
   console.log(req.params.id);
```

Comment:
yay you already have the start of this logic back here. Now in the front end you would do a fetch for the backend at backendURL/Notes/delete/:id with the proper logic up on the front end to get that id of the note you want to get rid of. 

You also might double check to see if you can use vars before they are declared. That deleteFunc might need to be declared before you actually use it.

Reply:
So, because I have patched things together and have fragments of different approaches I’m sabotaging myself?

Sounds like I need to look at App.js and Server.js side by side and move all routes to server.js or refactor, as you suggest, to a `notesController.js` file?